examples/topics/stats/density.py

```diff
@@ -0,0 +1,61 @@
+''' A `multiple kernel density estimation`_ plot of the "cows" data using
+the `sklearn.neighbors.KernelDensity`_ function and `Bokeh` varea glyph
+
+.. bokeh-example-metadata::
+    :sampledata: cows
+    :apis: bokeh.plotting.figure.varea
+    :refs: :ref:`ug_topics_stats_density`
+    :keywords: density, varea
+
+.. _kernel density estimation: https://en.wikipedia.org/wiki/Kernel_density_estimation
+.. _sklearn.neighbors.KernelDensity: https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KernelDensity.html
+
+'''
+import numpy as np
+from sklearn.neighbors import KernelDensity
+
+from bokeh.models import ColumnDataSource, Label, PrintfTickFormatter
+from bokeh.palettes import Dark2
+from bokeh.plotting import figure, show
+from bokeh.sampledata.cows import data as df
+
+grouped = df.groupby('breed')
+colors = Dark2[5]
+x = np.linspace(2, 8, 1000)
+source = ColumnDataSource(dict(x=x))
+
+p = figure(title="Multiple density estimates", height=300, x_range=(2.5, 7.5), x_axis_label="butterfat contents", y_axis_label="density")
+
+for i, (breed, group_df) in enumerate(grouped):
```

Comment:
Just for consideration, I think the enumeration and explicit loop index could be avoided with:
```python
for (breed, breef_df), color in zip(grouped, colors):
```
in any case I think think `breed_df` is a a bit more descriptive than `group_df` (and maybe `grouped` could be something like `by_breed` or `breed_groups`?)